### ISSUE TITLE
object's full path should include zone name component

### ISSUE THREAD

**Issue by d-w-moore** [hechth/fs-irods#16]
Objects in the catalog are known and queried by logical path, ie the name, ie:
```
ICAT=> select * from r_coll_main;
ICAT=> select coll_name from r_coll_main;
          coll_name          
-----------------------------
 /
 /tempZone
 /tempZone/home
 /tempZone/trash
 /tempZone/home/public
 /tempZone/trash/home/public
 /tempZone/trash/home/rods
 [...]
```
It seems, in keeping with this, that getting the path of a given collection,etc, via an iRODSFS should yield the same
However, we see this:
```
>>> irfs = fs_irods.iRODSFS(s)
<fs_irods.iRODSFS.iRODSFS object at 0x7f6d76843910>
>>> for path,_,_ in irfs.walk: print(path)
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'BoundWalker' object is not iterable
>>> for path,_,_ in irfs.walk(): print(path)
... 
/
/home
/trash
/home/alice
/home/public
/home/rods
/trash/home
/home/rods/dir
   [...]
```
The zone-named leading component of the paths are missing.

**Comment by trel** [comment]
Agreed, not using the zone name in the path means visibility across federation is off the table.

**Comment by d-w-moore** [comment]
There is also [this line ](https://github.com/hechth/fs-irods/blob/3e717e9aced68c33e62f938b8139ba8240d0090a/fs_irods/iRODSFS.py#L34) which actually examines the path's content (its leading element in particular) in order to tell whether or not it should be considered absolute.  That probably shouldn't be so. It is after all technically possible for a` /tempZone/tempZone` collection to exist.  So the true meaning of "/myZone" at best is ambiguous, but in some cases could resolve incorrectly.

Because of this, we get e.g. the erroneous result (in a default install of zone name `tempZone`)
```
$ imkdir -p /tempZone/{someDir,tempZone}/subDir
$ ils  /tempZone/{someDir,tempZone}/subDir
/tempZone/someDir/subDir:
/tempZone/tempZone/subDir:
$ python
>>> import fs_irods, irods.test.helpers
>>> ses = irods.test.helpers.make_session()
>>> f = fs_irods.iRODSFS(ses)
<fs_irods.iRODSFS.iRODSFS object at 0x7fe2d86bfb50>
>>> f.isdir('/someDir/subDir')
True
>>> f.isdir('/tempZone/subDir')
False
```
Yet another reason why I see it as desirable to unify the `fs-irods` notion of what a logical path is, with that of iRODS itself.  Any opinion on this, @hechth ?

**Comment by d-w-moore** [comment]
Taking a fresh look at this issue, I'm wondering if it's been partly misguided, ... mainly because I think the spirit of PyFilesystem2 is to present like a mount point, at least when taking the example of its use with a plain POSIX OS.  In other words the fs.osfs.OSFS class  takes a first initializer argument of a starting directory, and from then on, its internal "filesystem" is relative to that point in the overall Linux filesystem.  So if a file ~/dust/wind/dude exists, then OSFS("~/dust").isfile("/wind/dude") evaluates as True, right?
